canonicalwebteam/flask_base/app.py

```diff
@@ -17,32 +21,64 @@ def __init__(
         self,
         name,
         service,
-        prepare_deleted=prepare_deleted(),
```

Comment:
By removing `prepare_deleted` and `prepare_redirects` we remove the possibility of customizing the deleted callback and to set/unset permanent redirects.

Reply:
This was very deliberate - I'd rather that each of the sites use the exact same interface for processing redirects and deleted pages - as in, they keep them in `redirects.yam` and `deleted.yaml` respectively. Allowing the function to be customised encourages sites to customise it.

I'd be more than happy to look for a `permanent-redirects.yaml` in this default configuration and parse permanent redirects from there, but I think it's better not to offer the ability for sites to customise their implementation.

Reply:
I see your point, but with them being optional I don't think we are encouraging overriding. In any case the point was more about will we need permanent redirects or customized callbacks? I would be happy with a `permanent-redirects.yaml` or even tweaking `yaml-responses` so the status code could be set in `redirects.yaml`? Also happy to merge this as is and address this in a different PR.

Reply:
I've made it look for `permanent-redirects.yaml` as well. Is that okay?